src/sbr/SR.java

```diff
@@ -314,54 +314,58 @@ public GraphRestrictions restrictions() {
 
 	public void setrestrictions() {
 		for (Segment segment : segments) {
-			if (segment.getLinks().isEmpty())
+			if (segment.getLinks().isEmpty()) {
 				continue;
-
+			}
 			if (segment.isUnitary()) {
 				// No traffic allowed at link
 				Vertex Starting = segment.getLinks().get(0).source();
 				Vertex Ending = segment.getLinks().get(0).destination();
-				//System.err.println("Start: " + Starting.name() + " Ending: "+ Ending.name());
 				// Restricted link
-				String opStarting = Starting.edge(Ending).color();
-				String opEnding = Ending.edge(Starting).color();
+				Set<Character> oPStarting = new HashSet<>();
+				oPStarting.add(Starting.edge(Ending).color());
+				Set<Character> oPEnding = new HashSet<>();
+				oPEnding.add(Ending.edge(Starting).color());
 				// Restrictions at Starting core
-				for (Edge link : Starting.adjuncts())
-					if (link.color() != opStarting) {
-						restrictions.addRestriction(Starting, link.color(), opStarting);
-					}
+				for (Edge link : Starting.adjuncts()) {
+						restrictions.addRestriction(Starting, link.color(), oPStarting);
```

Comment:
addRestriction always receive only one restriction, so you don't need to receive a Set for 'ip' and 'op' (if you needed, you should call it addRestrictions). Making it receive a Character will simplify the setrestrictions method.